### PR TITLE
feat(shared): Improve useAssertWrappedByClerkProvider messaging

### PR DESCRIPTION
**Pull request by jacekradko**
## Description

Improving messaging coming from `useAssertWrappedByClerkProvider`

Related: SDKI-786

## Checklist

- [ ] `pnpm test` runs as expected.
- [ ] `pnpm build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

**Review by alexcarpenter — APPROVED** [review]

**Review by tmilewski — APPROVED** [review]

File: .changeset/thick-guests-burn.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@clerk/shared': patch
+---
+
+Improving error messaging from useAssertWrappedByClerkProvider to point out potential failures scenarios
```
Comment (BRKalow) [inline .changeset/thick-guests-burn.md:5]:
```suggestion
Improving error messaging from `useAssertWrappedByClerkProvider` to point out potential failure scenarios.
```

File: packages/shared/src/react/contexts.tsx [hunk 2]
```diff
@@ -71,7 +71,13 @@ function useAssertWrappedByClerkProvider(displayNameOrFn: string | (() => void))
     }
 
     throw new Error(
-      `${displayNameOrFn} can only be used within the <ClerkProvider /> component. Learn more: https://clerk.com/docs/components/clerk-provider`,
+      `Error: ${displayNameOrFn} can only be used within the <ClerkProvider /> component.
```
Comment (BRKalow) [inline packages/shared/src/react/contexts.tsx:74]: I think an error object will prefix with `Error:` by default? In any case, I don't think we need it.
```suggestion
      `${displayNameOrFn} can only be used within the <ClerkProvider /> component.
```

File: packages/shared/src/react/contexts.tsx [hunk 3]
```diff
@@ -71,7 +71,13 @@ function useAssertWrappedByClerkProvider(displayNameOrFn: string | (() => void))
...
+
+Possible causes:
```
Comment (BRKalow) [inline packages/shared/src/react/contexts.tsx:76]:
```suggestion
Possible fixes:
```

File: packages/shared/src/react/contexts.tsx [hunk 4]
```diff
@@ -71,7 +71,13 @@ function useAssertWrappedByClerkProvider(displayNameOrFn: string | (() => void))
...
+1. Ensure that the <ClerkProvider /> is correctly wrapping your application where this component is used.
+2. Check for multiple versions of the \`@clerk/shared\` package in your project, which could cause context mismatches. You can use a tool like \`npm ls @clerk/shared\` or \`pnpm list @clerk/shared\` to identify duplicates.
```
Comment (BRKalow) [inline packages/shared/src/react/contexts.tsx:78]: Let's make this more concise, we don't need to worry about multiple package managers for now.

```suggestion
2. Check for multiple versions of the \`@clerk/shared\` package in your project. Use a tool like \`npm ls @clerk/shared\` to identify multiple versions, and update your dependencies to only rely on one.
```

File: packages/shared/src/react/contexts.tsx [hunk 5]
```diff
@@ -71,7 +71,13 @@ function useAssertWrappedByClerkProvider(displayNameOrFn: string | (() => void))
...
+
+For more details on setting up the <ClerkProvider />, visit: https://clerk.com/docs/components/clerk-provider`.trim(),
```
Comment (BRKalow) [inline packages/shared/src/react/contexts.tsx:80]: I think it's okay to keep this concise too:
```suggestion
Learn more: https://clerk.com/docs/components/clerk-provider`.trim(),
```

**Review by BRKalow — CHANGES_REQUESTED** [review]

**Review by BRKalow — APPROVED** [review]

**Pull request merged**